pylint/checkers/utils.py

```diff
@@ -1963,3 +1963,71 @@ def is_hashable(node: nodes.NodeNG) -> bool:
         return False
     except astroid.InferenceError:
         return True
+
+
+def _is_target_name_in_binop_side(
+    target: nodes.AssignName | nodes.AssignAttr, side: nodes.NodeNG | None
+) -> bool:
+    """Determine whether the target name-like node is referenced in the side node."""
+    if isinstance(side, nodes.Name):
+        if isinstance(target, nodes.AssignName):
+            return target.name == side.name  # type: ignore[no-any-return]
+        return False
+    if isinstance(side, nodes.Attribute):
+        if isinstance(target, nodes.AssignAttr):
+            target_parent = target.parent
+            target_string = target.attrname or ""
+            while isinstance(target_parent, (nodes.Attribute, nodes.Name)):
+                if isinstance(target_parent, nodes.Attribute):
+                    target_string = f"{target_parent.attrname}.{target_string}"
+                else:
+                    target_string = f"{target_parent.name}.{target_string}"
+                target_parent = target_parent.parent
+
+            side_parent = side.parent
+            side_string = side.attrname or ""
+            while isinstance(side_parent, (nodes.Attribute, nodes.Name)):
```

Comment:
Perhaps these two`while` blocks could be extracted into a reusable function. They're both quite similar.

Reply:
Done!

Reply:
👍